### ISSUE TITLE
Export scene point cloud by region and align

### ISSUE THREAD

**Issue by Bin-ze** [nerfstudio-project/nerfstudio#1719]
I want to use nerfstudio to train different areas of a scene, and stitch the point clouds of different areas according to its pose.
But I can't determine the correspondence between the origin pose of the point cloud exported by nerfstudio and the pose of the image I provided. And I found that the point cloud is normalized into a box space, but I only found its clipping operation, but not its scaling operation, like this:
<img width="834" alt="image" src="https://user-images.githubusercontent.com/72087870/231041141-384b2b83-9705-4ddd-927f-9398cfff3f43.png">

In addition, I found that the origin of the exported point cloud is defined by nerfstudio itself, like this:
<img width="587" alt="image" src="https://user-images.githubusercontent.com/72087870/231041344-3b2d93d5-9fa3-476a-811a-1927262e1b37.png">
The origin of the coordinate system is the origin of the exported point cloud.

I would like to know how to achieve the above purpose?

**Comment by tancik** [comment]
Take a look at the `dataparser_transform.json` file in the outputs directory. It has info regard the transform applied during the training.

**Comment by LHXhh** [comment]
@Bin-ze Hello, I had a question to consult. Now I have an input file called transforms.json, in which each image poses a real camera external parameter in the world coordinate system.  Now I just get the output point cloud for nerfstudio's custom coordinate system and the dataparser_transform.json mentioned above.  
What do I have to do to get the point cloud that restores the real scene, that is, the point cloud whose origin is determined by the real external parameters in world coordinates.  Thank you very much.

**Comment by Bin-ze** [comment]
> @Bin-ze Hello, I had a question to consult. Now I have an input file called transforms.json, in which each image poses a real camera external parameter in the world coordinate system. Now I just get the output point cloud for nerfstudio's custom coordinate system and the dataparser_transform.json mentioned above. What do I have to do to get the point cloud that restores the real scene, that is, the point cloud whose origin is determined by the real external parameters in world coordinates. Thank you very much.

dataparser_transform.json stores the transformation from original space to nerf studio space, and the point cloud you export is in nerf studio space. In order to get the point cloud in the original space, you need to inverse transform the exported point cloud, using the scaling factor and transformation matrix saved in dataparser_transform.json

**Comment by LHXhh** [comment]
Thank you very much!

**Comment by 19521242bao** [comment]
> > @Bin-ze Hello, I had a question to consult. Now I have an input file called transforms.json, in which each image poses a real camera external parameter in the world coordinate system. Now I just get the output point cloud for nerfstudio's custom coordinate system and the dataparser_transform.json mentioned above. What do I have to do to get the point cloud that restores the real scene, that is, the point cloud whose origin is determined by the real external parameters in world coordinates. Thank you very much.
> 
> dataparser_transform.json stores the transformation from original space to nerf studio space, and the point cloud you export is in nerf studio space. In order to get the point cloud in the original space, you need to inverse transform the exported point cloud, using the scaling factor and transformation matrix saved in dataparser_transform.json


what  formulation used for inverse ?

**Comment by Bin-ze** [comment]
> nerf_sturio_pcd /= scale_factor
> nerf_studio_pcd = (torch.cat(nerf_studio_pcd, torch.ones(nerf_studio_pcd.shape[0])) * torch.inverse(transform))[:, :3]

The general transformation logic is like this, it may need to be adjusted in detail
